infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InitializedState.java

```diff
@@ -78,7 +78,19 @@ public void onEntry(LogReplicationState from) {
     @Override
     public void onExit(LogReplicationState to) {
         if (to != this || to.getType() != LogReplicationStateType.ERROR) {
-            fsm.getAckReader().startSyncStatusUpdatePeriodicTask();
+            switch (to.getType()) {
+                case IN_LOG_ENTRY_SYNC:
+                    fsm.getAckReader().startSyncStatusUpdatePeriodicTask(LogReplicationMetadata.ReplicationStatusVal.SyncType.LOG_ENTRY);
+                    break;
+                case IN_SNAPSHOT_SYNC:
+                    fsm.getAckReader().startSyncStatusUpdatePeriodicTask(LogReplicationMetadata.ReplicationStatusVal.SyncType.SNAPSHOT);
+                    break;
+                case WAIT_SNAPSHOT_APPLY:
+                    fsm.getAckReader().startSyncStatusUpdatePeriodicTask(LogReplicationMetadata.ReplicationStatusVal.SyncType.SNAPSHOT);
+                    break;
+                default:
```

Comment:
+1, I think it's better to do the `setSyncType(type)` in processEvent and leave the `startSyncStatusUpdatePeriodicTask()` as it was. 

Also, looks like the `setSyncType()` on the entry of InLogEntrySyncState could be removed now to avoid unnecessary lock acquire.

Reply:
Good finding. I updated the logic.